lexibank_hubercolumbian.py

```diff
@@ -1,83 +1,60 @@
         args.writer.add_sources()
 
         # get the language identifiers stored in wl._meta['doculect'] parsed from input file
-        lids = {}
-        for line in wl._meta["doculect"]:
-            rest = line.split(", ")
-            name = rest.pop(0)
-            lids[name] = rest.pop(0)
-
-        concept_lookup = {}
-
-        for concept in self.conceptlists[0].concepts.values():
-            concept_id = "%s_%s" % (concept.number, slug(concept.english))
-
-            args.writer.add_concept(
-                ID=concept_id,
-                Name=concept.english,
-                Concepticon_ID=concept.concepticon_id,
-                Concepticon_Gloss=concept.concepticon_gloss,
-                Gloss_in_digital_source=concept.attributes["gloss_in_digital_source"],
-                Spanish=concept.attributes["spanish"],
-            )
-
-            concept_lookup[concept.attributes["gloss_in_digital_source"]] = concept_id
-
-        def grouped_rows(wl):
-            rows = [
-                (
-                    wl[k, "counterpart_doculect"],
-                    wl[k, "concept"],
-                    wl[k, "counterpart"],
-                    wl[k, "qlcid"],
-                )
-                for k in wl
-                if wl[k, "counterpart_doculect"] not in ["English", "Español"]
-            ]
-            return groupby(sorted(rows), key=lambda r: (r[0], r[1]))
-
-        for (language, concept), rows in progressbar(grouped_rows(wl)):
-            iso = lids[language]
-
-            args.writer.add_language(
-                ID=slug(language, lowercase=False),
-                Name=language,
-                ISO639P3code=iso,
-                Glottocode=self.glottolog.glottocode_by_iso.get(iso, ""),
-            )
-
-            for i, (l, c, form, id_) in enumerate(rows):
-                args.writer.add_form(
-                    Language_ID=slug(language, lowercase=False),
+        language_lookup = args.writer.add_languages(
+                lookup_factory="Name_in_Source")
+
+        concept_lookup = args.writer.add_concepts(
+                id_factory=lambda x: x.id.split('-')[-1]+'_'+slug(x.english),
+                lookup_factory='Name_in_Source')
+        
+        rows = [(doculect, concept, value, qlcid) for (
+            idx, doculect, concept, value, qlcid) in wl.iter_rows(
+            'counterpart_doculect',
+            'concept',
+            'counterpart',
+            'qlcid'
+            ) if doculect
+                not in ['English', 'Español']]
+
+        for doculect, concept, value, qlcid in rows:
+            args.writer.add_form(
+                    Language_ID=language_lookup[doculect],
                     Parameter_ID=concept_lookup[concept],
                     Value=form,
                     Form=form,
-                    Source=["Huber1992"],
-                    Local_ID=id_,
-                )
+                    Source=['Huber1992'],
+                    Local_ID=qlcid
```

Comment:
The local ID is something I myself have to check against. As this would mean, that I could use all datasets like STEDT, and assign them the local id, which is to be preferred over a `CustomLexeme` class. Again, something we should maybe emphasize in our tutorial: if the original dataset has its own ID, place it into local_id. This is essential in cases like STEDT or other providers, as we want to embrace their work but also be able to trace it back to their sources.